lib/ansible/modules/network/f5/bigip_irule_bigsuds.py

```diff
@@ -0,0 +1,263 @@
+    validate_certs = module.params['validate_certs']
+
+    state = module.params['state']
+    name = fq_name(partition, module.params['name']) # Fully Qualified name (including the partition)
+
+    # Irule contents can either be defined inline via 'content' attribute, or by passing the path to 
+    # a file via 'src' attribute, for the latter we need to read those contents from the file.
+    content = None
+    if module.params['src']:
+        try:
+            with open(module.params['src']) as f:
+                content = f.read()
+        except Exception as e:
+            raise Exception('Error reading iRule "src" file : %s' % e)
+    else:
+        content = module.params['content']
+
+    try:
+        api = bigip_api(server, user, password, validate_certs, port=server_port)
+        result = {'changed': False, 'irule_name': name}  # default module return value
+
+        if state == 'absent':
+            # Check mode is disabled
+            if not module.check_mode:
+                if irule_exists(api, name):
+                    try:
+                        irule_remove(api, name)
+                        result = {'changed': True, 'action': 'deleted'}
+                    except bigsuds.OperationFailed as e:
+                        # Handles the situation where the irule was deleted in between us querying for its existence and running the delete command.
+                        if "was not found" in str(e):
+                            result['changed'] = False
+                        else:
+                            raise
+            # Check mode is enabled
+            else:
+                result = {'changed': True}
+
+        # State is 'present'
+        else:
+            # Check mode is disabled
+            if not module.check_mode:
+                # If the irule doesn't exist we can create it.
+                if not irule_exists(api, name):
+                    try:
+                        irule_create(api, name, content)
+                        result = {'changed': True, 'action': 'created'}
+                    except Exception as e:
+                        raise Exception('Error creating iRule : %s' % e)
+               
+                # The irule already exists so we need to check if it has the correct content
+                # and update it only if necessary, so Ansible can report 'changed' correctly.
+                else:
+                    try:
+                        result = irule_update(api, name, content, module._diff)
+                    except Exception as e:
+                        raise Exception("Error updating iRule : %s" % e)
+
+            # Check mode is disabled
+            else:
+                # check-mode return value
+                result = {'changed': True}
+
+    except Exception as e:
+        module.fail_json(msg="Received exception: %s" % e)
+
+    module.exit_json(**result)
+
+# import module snippets
+from ansible.module_utils.basic import *
```

Comment:
1) Import what you need, rather than `*`
2) Imports should go after RETURN

https://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html#python-imports